### PR TITLE
Use Package repo 2.0 

### PR DESCRIPTION
**Pull request by Vincinator**
This commit transitions our Garden Linux package lifecycle management from a GitLab-centric to a container-centric approach. The previous design, heavily reliant on GitLab, did not align with our requirements and demanded excessive manual maintenance. By adopting this new strategy, we generate the apt repository through a more efficient container-centric process, streamlining operations and reducing the need for manual intervention.

For more details about the new repository creation, please checkout the following repositories:
- https://github.com/gardenlinux/repo
- https://github.com/gardenlinux/package-build

### CODE REVIEW

**Comment by Vincinator** [comment]
moved arm64 dependent setup code to setup actions. Need to add parameter "amd64" to #1994

File: hack/gl-search.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ if [ "$gls_gl_dist" != "today" ]; then
   fi
 fi  
 
-repo_url="$base_url/dists/${gls_gl_dist}/Release?ignoreCaching=1"
+repo_url="$base_url/dists/${gls_gl_dist}/InRelease?ignoreCaching=1"
```
Comment (nkraetzschmar) [inline hack/gl-search.sh:49]: we can probably get rid of the `ignoreCaching` pseudo query part, as the new repo pipeline explicitly invalidates all caches when needed (https://github.com/gardenlinux/repo/blob/main/.github/workflows/build.yml#L129)

File: build [hunk 2]
```diff
@@ -6,7 +6,7 @@ shopt -s nullglob
 exec 3>&1
 exec 1>&2
 
-container_image=localhost/builder
+container_image=ghcr.io/gardenlinux/builder:194d7536f837076677727d890c5a16d2dd24ef66
```
Comment (nkraetzschmar) [inline build:9]: this should be `localhost/builder` to keep the features introduced by #1874

**Review by nkraetzschmar — APPROVED** [review]
lgtm

**Pull request merged**